config/set/downgrade.php

```diff
@@ -2,11 +2,17 @@
 
 declare(strict_types=1);
 
+use Rector\Downgrade\Rector\FunctionLike\DowngradeParamObjectTypeDeclarationRector;
+use Rector\Downgrade\Rector\FunctionLike\DowngradeReturnObjectTypeDeclarationRector;
 use Rector\Downgrade\Rector\Property\DowngradeTypedPropertyRector;
 use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
 
 return static function (ContainerConfigurator $containerConfigurator): void {
     $services = $containerConfigurator->services();
 
+    $services->set(DowngradeParamObjectTypeDeclarationRector::class);
+
+    $services->set(DowngradeReturnObjectTypeDeclarationRector::class);
```

Comment:
I wonder, why keeping them separate? How is it useful to run just one without the other?

Reply:
Hmmmm, that's a good point. I guess they are always executed together.

But even if that's the case, I wonder if there is value in keeping them separate:

- Each rule can have its own, more specific tests (eg: the test `multiple_params.php.inc` doesn't make sense for return types)
- It makes code more SOLID

If this makes sense, then you could possibly have them shipped through a set `downgrade-object`, so that they are still always executed together, and then have sets execute not only rules but also other sets, so that set `downgrade` executes set `downgrade-object`. Is this possible?

Otherwise, I'll combine the 2 rules into 1: `DowngradeObjectTypeDeclarationRector`.

Please let me know what you prefer

Reply:
Actually, yes, there is a use case: In PHP 8.0, `static` is a return type, but not a param type: https://stitcher.io/blog/new-in-php-8#new-static-return-type-rfc

Reply:
Keeping it decoupled, the rule to remove return type `static` can use the same abstract class as for return types `object` and `mixed`

Reply:
Thanks for detailed explanation. Let's go with your approach. I believe it will bring more value on additional type rules

Reply:
Which approach? What is already implemented, so I don't do anything? Or create a set `downgrade-object` and execute it within set `downgrade`?

Reply:
> What is already implemented, so I don't do anything?

:+1: